oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticIndexPlanner.java

```diff
@@ -48,6 +53,8 @@ protected List<QueryIndex.OrderEntry> createSortOrder(IndexDefinition.IndexingRu
             PropertyDefinition pd = rule.getConfig(propName);
             if (pd != null
                     && o.getPropertyType() != null
+                    // functions on regexp-based properties must be skipped since the values cannot be indexed
+                    && (!pd.isRegexp || !propName.startsWith(QueryConstants.FUNCTION_RESTRICTION_PREFIX))
```

Comment:
Do we want to ignore those properties that have regex enabled and have a function defined or either of these ? As per the comment it should be the first one but the condition is for the latter.